reconcile/test/fixtures/oc_native/api.yml

```diff
@@ -0,0 +1,98 @@
+server:
+  version:
+    major: 1
+  api/v1:
+    kind: APIResourceList
+    resources: 
+    - name: pods
+      kind: Pod
+    - name: pods/status
+      kind: Pod
+    - name: services
+      kind: Service
+  apis:
+    kind: APIGroupList
+    groups:
+    - name: project.openshift.io
+      versions:
+      - groupVersion: project.openshift.io/v1
+        version: v1
+      preferredVersion:
+        groupVersion: project.openshift.io/v1
+        version: v1
+    - name: config.openshift.io
+      versions:
+      - groupVersion: config.openshift.io/v1
+        version: v1
+      preferredVersion:
+        groupVersion: config.openshift.io/v1
+        version: v1
+    - name: upgrade.managed.openshift.io
+      versions:
+      - groupVersion: upgrade.managed.openshift.io/v1alpha1
+        version: v1alpha1
+      preferredVersion:
+        groupVersion: upgrade.managed.openshift.io/v1alpha1
+        version: v1alpha1
+    - name: managed.openshift.io
+      versions:
+      - groupVersion: managed.openshift.io/v1alpha1
+        version: v1alpha1
+      - groupVersion: managed.openshift.io/v1alpha2
+        version: v1alpha2
+      preferredVersion:
+        groupVersion: managed.openshift.io/v1alpha2
+        version: v1alpha2
```

Comment:
I would add 2 more items here:

- Order so non-preferred version is first
- Multiple versions, but no preferred version

Reply:
Add the fist one
the preferred version can not be empty